### ISSUE TITLE
confused about logL, score and score_samples

### ISSUE THREAD

**Issue by dbkeitel** [tholoien/XDGMM#37]
Dear developers, I've just discovered this package and am trying to use it for some simple how-many-components GMM model selection on data with individual measurement errors on each data point, which it seems the more commonly known sklearn.mixture cannot do. My tests look promising in terms of visual fit quality and getting BICs consistent with my human intuition.

But first, I wanted to make sure I fully understand the outputs in the even simpler case assuming equal errors. Where I'm struggling is in interpreting the likelihood/scores outputs of a fitted xdgmm object. As I understand it, score_samples() should give the likelihood of each data point, score() should give the average over all those, and logL() the total. And indeed it seems that the following all agree with each other _for a single component_:
xdgmm.score_samples(X,Xerr).sum()
len(X)*xdgmm.score(X,Xerr)
xdgmm.logL(X,Xerr)
They also agree with len(X)*gmm.score(X) from sklearn.mixture.GaussianMixture.
However, _for a multi-component model_, the sum of score_samples() still (roughly) agrees with sklearn, but score() and logL()

 yield a very different result.

Am I just missing some number-of-parameters-dependent normalisation, or is there something more going on?

To demonstrate, I've made a [simple example notebook](https://nbviewer.jupyter.org/gist/dbkeitel/d6dbb0b13ad2d4f587f56f70f721ebee) where the dataset is a trimmed-down 1D version of the one from the package's Demo.ipynb. The key outputs are e.g. "sumscores=-237.13, len*score=-167.82, logL=-167.82" for a c=2 GMM in cell 6.

Thanks for your time!

**Comment by dbkeitel** [comment]
This actually seems to be a bug in astroML that was fixed back in 2015 by [this commit](https://github.com/astroML/astroML/commit/a372bfc2712d56b7e0bff62e6ea4d452e61adcb8); but since the [last pypi release](https://pypi.org/project/astroML/) is dated Jan 28, 2015 this fix never made it. If I rerun mit linked example notebook with an install from github, then sumscores,  len*score and logL all agree as expected.

@tholoien Would it be worth pointing out in the XDGMM documentation that getting a more recent astroML from github is recommended?

**Comment by tholoien** [comment]
Thanks for pointing that out. I have updated the README to note this. 

Closing issue.
